### PR TITLE
EZEE-2880: Custom workflow actions

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZEE-2880
| Versions      | master

To do:
- [x] Add example for an action triggered on transition, with setting context and/or returning a result

### CODE REVIEW

File: docs/guide/extending/extending_workflow.md [hunk 1]
```diff
@@ -1,5 +1,166 @@
...
+        $message = $metadata['data']['message'];
+
+        $this->notificationHandler->info(
+            $message,
+            [],
+            'domain'
+        );
+
+        $this->setResult($event, true);
+
```
Comment (Nattfarinn) [inline docs/guide/extending/extending_workflow.md]: Unnecessary empty line

File: docs/guide/extending/extending_workflow.md [hunk 2]
```diff
@@ -1,5 +1,165 @@
 # Extending Workflow
 
+## Adding custom actions
+
+[Built-in actions in the Editorial Workflow](../workflow.md#publishing-content-with-workflow)
+enable you to automatically publish a Content item or to sent a notification to reviewers.
```
Comment (MagdalenaZuba) [inline docs/guide/extending/extending_workflow.md]:
```suggestion
enable you to automatically publish a Content item or to send a notification to reviewers.
```

File: docs/guide/extending/extending_workflow.md [hunk 3]
```diff
@@ -1,5 +1,165 @@
...
+            to: proofread
+            label: To proofreading
+            actions:
+                proofread_transition_action:
+                    data:
+                        message: "The article has gone to proofreading"
+```
+
+The configuration indicates the name of the custom action (`proofread_transition_action`) that will call the action.
+`data` contains additional data that can be passed to the action. In this case it a message to display.
```
Comment (MagdalenaZuba) [inline docs/guide/extending/extending_workflow.md]:
```suggestion
`data` contains additional data that can be passed to the action. In this case, a message to display.
```

File: docs/guide/extending/extending_workflow.md [hunk 4]
```diff
@@ -1,5 +1,165 @@
...
+
+To define what the action does, create an Event Listener, in this case `src/EventListener/ProofreadTransitionListener.php`:
```
Comment (MagdalenaZuba) [inline docs/guide/extending/extending_workflow.md]:
```suggestion
To define what the action does, create an Event Listener `src/EventListener/ProofreadTransitionListener.php`:
```

File: docs/guide/extending/extending_workflow.md [hunk 5]
```diff
@@ -1,5 +1,165 @@
...
+
+        $this->setResult($event, true);
+    }
+}
+```
+
+This Listener displays a notification bar at the bottom of the page when a Content item goes through the `to_proofread` transition.
+
+The content of the notification is the message configured in `actions.proofread_transition_action.data`.
+To get it, you access the metadata for this transition through `getActionMetadata()` (line 27).
```
Comment (MagdalenaZuba) [inline docs/guide/extending/extending_workflow.md]:
```suggestion
To get it, you need to access the metadata for this transition through `getActionMetadata()` (line 27).
```
OR
```suggestion
To get it, access the metadata for this transition through `getActionMetadata()` (line 27).
```

File: docs/guide/extending/extending_workflow.md [hunk 6]
```diff
@@ -1,5 +1,165 @@
...
+This listener also displays a notification, but in this case its content is taken from the message
+that the user types when choosing the `Done` transition.
+
+The message is contained in the context of the action.
+
+`$event->getContext()` gives you access to the context.
+The context contains:
+
+- `$workflowId` - the ID of the current workflow
+- `$message` - content of the message input by the user when sending the Content item through the transitions
```
Comment (MagdalenaZuba) [inline docs/guide/extending/extending_workflow.md]:
```suggestion
- `$message` - content of the user's message when sending the Content item through the transitions
```

**Pull request merged**